register_service/_crypto.py

```diff
@@ -0,0 +1,222 @@
+import collections
+import os
+import io
+import math
+import struct
+
+from nacl.bindings import crypto_scalarmult_base, crypto_scalarmult
+
+from cryptography.exceptions import InvalidTag
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from cryptography.hazmat.primitives.hashes import SHA512
+from cryptography.hazmat.primitives.hmac import HMAC
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+
+# NOTE: All assert's here are logical asserts and do NOT depend on input data. I.e. they only explicate the contract.
+#       Error handling on the other hand uses NoiseError and other crypto exceptions.
+
+backend = default_backend()
+
+
+SUITE_NAME = b"Noise255/AES256-GCM\0\0\0\0\0"
+H_LEN = 64
+CV_LEN_BYTE = 48
+SYMM_KEY_LEN_BYTE = 32
+NONCE_LEN_BYTE = 12
+MAC_BYTES = 16
+HEADER_CIPHER_TEXT_LEN_BYTE = 48
+PADDING_LEN_BYTES = 4
+ASYM_KEY_SIZE_BYTE = 32
+
+
+DecryptedNoiseBox = collections.namedtuple('DecryptedNoiseBox', 'sender_pubkey contents')
+
+
+class KeyPair:
+    def __init__(self, private_key=None, public_key=None):
+        if not private_key:
+            private_key = os.urandom(ASYM_KEY_SIZE_BYTE)
+        if isinstance(private_key, str):
+            private_key = decode_key(private_key)
+        assert len(private_key) == ASYM_KEY_SIZE_BYTE
```

Comment:
assertions are ignored if python is started with -O, better make it an if statement